src/Arcus.Messaging.Pumps.Abstractions/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -35,5 +35,52 @@ public static IServiceCollection AddMessagePump<TMessagePump>(
             
             return services.AddHostedService(implementationFactory);
         }
+
+        /// <summary>
+        /// Adds an <see cref="ICircuitBreakerEventHandler"/> implementation for a specific message pump to the application services.
+        /// </summary>
+        /// <typeparam name="THandler">The custom type of the event handler.</typeparam>
+        /// <param name="services">The application services to register the event handler.</param>
+        /// <param name="jobId">The unique ID to distinguish the message pump to register this event handler for.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="services"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="jobId"/> is blank.</exception>
+        public static IServiceCollection AddCircuitBreakerEventHandler<THandler>(this IServiceCollection services, string jobId)
+            where THandler : ICircuitBreakerEventHandler
```

Comment:
Is there a reason why the type-parameter here is called `THandler` whereas in the method below it is named `TEventHandler` ?